### PR TITLE
Use testing.T.Cleanup() in new cli test framework

### PR DESCRIPTION
**Pull request by tdeebswihart**
## What was changed
All manual calls to stop shared resources were replaced by calling `thing.Stop` as a cleanup function

## Why?
We don't need to track this ourselves, we can let the testing library do it for us

### CODE REVIEW

File: temporalcli/commands_test.go [hunk 1]
```diff
@@ -317,15 +293,9 @@ func StartDevServer(t *testing.T, options DevServerOptions) *DevServer {
 		require.NoError(t, err)
 	}
 
-	success = true
 	return d
 }
 
-func (d *DevServer) Stop() {
```
Comment (cretz) [inline temporalcli/commands_test.go:324]: If I choose to invoke `StartDevServer` I should be able to choose when to stop it. I don't think we should assume lifetimes of things we didn't start.

File: temporalcli/commands_test.go [hunk 2]
```diff
@@ -411,13 +381,10 @@ func (d *DevServer) StartDevWorker(t *testing.T, options DevWorkerOptions) *DevW
 	w.Worker.RegisterActivity(ops.DevActivity)
 	// Start worker or fail
 	require.NoError(t, w.Worker.Start(), "failed starting worker")
+	t.Cleanup(w.Worker.Stop)
 	return w
 }
 
-func (d *DevWorker) Stop() {
```
Comment (cretz) [inline temporalcli/commands_test.go:417]: Same here, if I as the caller choose to start the worker, I should be able to choose to stop it

File: temporalcli/commands_test.go (hunk 1 above)
Comment (Quinn-With-Two-Ns) [reply temporalcli/commands_test.go:324]: Yes, I need this functionality for tests I am adding

Comment (tdeebswihart) [reply temporalcli/commands_test.go:324]: Then I'll close this. Until your work it wasn't necessary :)

**Comment by tdeebswihart** [comment]
Closing as we actually will want to close these manually in future tests

**Pull request closed without merging**